Libraries/xcexecution/Sources/Executor.cpp

```diff
@@ -23,3 +23,23 @@ Executor::
 ~Executor()
 {
 }
+
+std::unordered_map<std::string, std::string> Executor::
+extendedEnvironmentVariables(
+    const std::vector<std::string> &executablePaths,
+    const std::unordered_map<std::string, std::string> &environmentVariables)
+{
+    if (environmentVariables.find("PATH") == environmentVariables.end()) {
+        // if PATH environment variable is not overriden,
+        // prepend current executablePaths
+        auto pathExtendedEnvironmentVariables = environmentVariables;
+        std::string pathEnvironmentVariable = "";
+        for (auto path : executablePaths) {
+            pathEnvironmentVariable += path + ":";
+        }
+        pathExtendedEnvironmentVariables["PATH"] = pathEnvironmentVariable + std::string(getenv("PATH"));
```

Comment:
Instead of `getenv()`, use `process::Context::environmentVariable()`. That allows simulating the environment for tests. (A `process::Context` is passed into this class.)